### ISSUE TITLE
false positive: use_g_strcmp0

### ISSUE THREAD

**Issue by blandford** [bilelmoussaoui/gobject-lint#32]
I have the following report in a sorting function:

```
    IndicatorListModelRow *row_a = (IndicatorListModelRow *) a;
    IndicatorListModelRow *row_b = (IndicatorListModelRow *) b;
  ...
    guint pop_a = indicator_list_model_row_get_popularity (row_a);
    guint pop_b = indicator_list_model_row_get_popularity (row_b);
  ...
    if (pop_a == pop_b)
  >    return g_strcmp0 (indicator_list_model_row_get_indicator (row_a),
                        indicator_list_model_row_get_indicator (row_b));
  ...
    return (pop_b - pop_a);
  error: g_strcmp0() returns 0 for equality — use 'g_strcmp0(...) == 0' or 'g_strcmp0(...) != 0' instead of bare boolean check
  rule: use_g_strcmp0
```
In this instance it is a comparison, and shouldn't return a boolean.
